### PR TITLE
Add .jenkins extension type

### PR DESCRIPTION
**Pull request by jhancock93**
Creates a .jenkins extension and filetype. Detects 'Jenkinsfile' and *.jenkins as that type.

### CODE REVIEW

File: identify/extensions.py [hunk 1]
```diff
@@ -247,6 +247,7 @@
     '.gitlint': EXTENSIONS['ini'] | {'gitlint'},
     '.gitmodules': {'text', 'gitmodules'},
     '.hgrc': EXTENSIONS['ini'] | {'hgrc'},
+    '.jenkins': EXTENSIONS['jenkins'],
```
Comment (asottile) [inline identify/extensions.py:250]: do people make jenkinsfiles that are just `.jenkins` -- I don't think this is necessary due to the extensions listed above -- I could be wrong here though (?)

Comment (jhancock93) [reply identify/extensions.py:250]: There are times where a repository might have multiple jenkins files. One for CI/CD purposes and one for deployment purposes. They can't all have the name jenkinsfile, in which case a .jenkins extension is a good convention for other pipelines.

Comment (asottile) [reply identify/extensions.py:250]: what I'm saying is this entry right here is redundant **unless** the filename is exactly `.jenkins` (with no name, just extension)

**Review by asottile — APPROVED** [review]

**Pull request merged**